src/data/unit_src.rs

```diff
@@ -0,0 +1,96 @@
+use af;
+use af::{Array, Dim4};
+use std::cell::{RefCell, Cell};
+
+use initializations;
+
+use data::{Data, DataSource, DataParams, Normalize, Shuffle};
+
+pub struct MultUniformSource {
+    pub params: DataParams,
+    pub iter: Cell<u64>,
+    pub offset: Cell<f32>,
+    pub bptt_unroll : u64, 
+}
+
+impl MultUniformSource {
```

Comment:
Btw, is this a good test of the Unitary RNN? It looks like it is just reconstructing noise? I am working on an XOR example that we can use for RNN tests. I.e. if x is a 1d signal and y is also 1d then it has to do the following:
```bash
x = 0 0 1 1
y = 0 0 1 0
```
 - Here if the previous x is 0 and the current x is 0 it requires y to be 1. 
 - If previous x is 1 and current x is 1 then y is 1
 - i.e. a basic xor source